docs/current_docs/api/internals.mdx

```diff
@@ -144,58 +144,152 @@ As this example demonstrates, Dagger object IDs hold the state of their correspo
+
+    ![Webhook.site request](/img/current_docs/api/webhook-2.png)
+
+1. Clone the Go examples repository and make it your working directory:
+
+    ```shell
+    git clone https://github.com/golang/example
+    cd example
+    ```
+
+    Next, execute the following query:
+
+    ```graphql
+    query {
+      host {
+        directory(path: "./hello") {
           id
         }
       }
     }
-  }
-}
-```
+    ```
 
-This query instructs Dagger to:
+    The return value of this query is an ID representing the state of the `hello` sub-directory of the repository.
 
-- initialize a container from the `alpine:latest` image (this resolves the `from` field);
-- add the `curl` command-line tool to the image via `apk add` (this resolves the `withExec` field);
-- send an HTTP request to your webhook URL using `curl` (this resolves the second `withExec` field);
-- return an ID representing the container state (this resolves the final `id` field).
+    Next, execute the following query, replacing the placeholder with the contents of the `id` field from the previous query:
 
-The query returns a base64-encoded block, as explained in the previous section.
+    ```graphql
+    query {
+      container {
+        from(address: "golang:latest") {
+          withDirectory(path: "/src", directory: "YOUR-ID-HERE") {
+            withWorkdir(path: "/src") {
+              withExec(args: ["go", "build"]) {
+                id
+              }
+            }
+          }
+        }
+      }
+    }
+    ```
 
-However, check the Webhook.site dashboard and you will notice that no HTTP request was sent when this query was executed. The reason is laziness: the query requests only an ID and, since resolving this does not require the commands to be executed, Dagger does not do so. It merely returns the container state and execution plan without actually executing the plan or running the `curl` command.
+    This query instructs Dagger to:
 
-Now, update and execute the query again as follows:
+    - initialize a container from the `golang:latest` image (this resolves the `from` field);
+    - copy the `hello` directory from the host to `/src` in the container (this resolves the `withDirectory` field);
+    - make the `/src/hello` directory in the container the current working directory (this resolves the `withWorkdir` field);
+    - build the source code in the working directory (this resolves the `withExec` field);
 
-```graphql
-query {
-  container {
-    from(address: "alpine:latest") {
-      withExec(args: ["apk", "add", "curl"]) {
-        withExec(args: ["curl", "YOUR-WEBHOOK-URL-HERE"]) {
-          stdout
+    However, when you run this query, you will notice that the binary is not actually built. Again, this is because the build result is not being used (neither executed or transferred back to the host), Dagger optimizes away that step.
```

Comment:
I feel like this needs more emphasis, maybe a note or important section

Reply:
Not sure about this, it seems like a normal sentence and paragraph.

Update: tried this, but having that block in the middle of the explanation looks weird, so dropping.